src/module_4/groceries_model_fitting.md

```diff
@@ -0,0 +1,1642 @@
+    plot_metrics(
+        f"RandomForestClassifier (n_estimators={tree_size})",
+        train_proba,
+        y_train,
+        (fig1, ax1),
+    )
+    val_proba = model.predict_proba(X_val[train_cols])[:, 1]
+    plot_metrics(
+        f"RandomForestClassifier (n_estimators={tree_size})",
+        val_proba,
+        y_val,
+        (fig2, ax2),
+    )
+    train_ap_, val_ap_, train_log_loss_, val_log_loss_ = calculate_metrics(
+        f"RandomForestClassifier (n_estimators={tree_size})",
+        train_proba,
+        y_train,
+        val_proba,
+        y_val,
+    )
+    train_log_loss.append(train_log_loss_)
+    val_log_loss.append(val_log_loss_)
+    train_avg_precision.append(train_ap_)
+    val_avg_precision.append(val_ap_)
+
+plot_metrics(
+    "Popularity baseline",
+    y_pred=val_df["global_popularity"],
+    y_test=val_df[target],
+    figure=(fig2, ax2),
+)
+
+fig, ax = plt.subplots(1, 2, figsize=(14, 4))
+ax[0].plot(n_estimators, train_log_loss, label="Train scores")
+ax[0].plot(n_estimators, val_log_loss, label="val scores")
+ax[0].plot(
+    n_estimators[np.argmin(val_log_loss)],
+    min(val_log_loss),
+    marker="o",
+    color="r",
+    label="min val score",
+)
+ax[0].set_ylabel("Cross entropy loss")
+ax[0].set_xlabel("n_estimators")
+ax[0].legend(loc="best")
+ax[0].set_title("Cross entropy loss vs n_estimators")
+
+ax[1].plot(n_estimators, train_avg_precision, label="Train scores")
+ax[1].plot(n_estimators, val_avg_precision, label="val scores")
+ax[1].plot(
+    n_estimators[np.argmax(val_avg_precision)],
+    max(val_avg_precision),
+    marker="o",
+    color="r",
+    label="max val score",
+)
+ax[1].set_ylabel("Average precision")
+ax[1].set_xlabel("n_estimators")
+ax[1].legend(loc="best")
+ax[1].set_title("Average precision vs n_estimators")
+```
+
+    RandomForestClassifier (n_estimators=5): Train AUC = 0.991, Val AUC = 0.645; Train AP = 0.770, Val AP = 0.072; Train Crossentropy = 0.025, Val Crossentropy = 0.345
+    RandomForestClassifier (n_estimators=25): Train AUC = 0.995, Val AUC = 0.714; Train AP = 0.852, Val AP = 0.111; Train Crossentropy = 0.021, Val Crossentropy = 0.251
+    RandomForestClassifier (n_estimators=50): Train AUC = 0.996, Val AUC = 0.731; Train AP = 0.859, Val AP = 0.119; Train Crossentropy = 0.021, Val Crossentropy = 0.222
+    RandomForestClassifier (n_estimators=100): Train AUC = 0.996, Val AUC = 0.749; Train AP = 0.863, Val AP = 0.125; Train Crossentropy = 0.021, Val Crossentropy = 0.190
+
+
+
+
+
+    Text(0.5, 1.0, 'Average precision vs n_estimators')
+
+
+
+
+    
+![png](groceries_model_fitting_files/groceries_model_fitting_18_2.png)
+    
+
+
+
+    
+![png](groceries_model_fitting_files/groceries_model_fitting_18_3.png)
+    
+
+
+
+    
+![png](groceries_model_fitting_files/groceries_model_fitting_18_4.png)
+    
+
+
+- RF with better peformance in all metrics(roc_AUC, PR_AUC, AP and CrossEntropy) are the ones with more trees. 
```

Comment:
As the theory suggests, so nice.

The cross entropy does not seem to have converged so maybe we can have checked with further trees